src/main/java/org/deepsymmetry/beatlink/MediaDetails.java

```diff
@@ -122,6 +120,34 @@ private int getUTF16StringLength(byte[] source, int offset, int maxLength) {
         return numBytes;
     }
 
+    /**
+     * Constructor sets all the immutable interpreted fields based on constructor inputs
+     *
+     * @param packet the media response packet that was received
+     */
+
+    /**
+     * Custom constructor to emulate an actual MediaDetails object from CDJs. This allows our Status packets
+     * to use OpusProvider to enrich the track data.
+     *
+     * @param slotReference Slot Reference to Emulate
+     * @param mediaType Media Type to Emulate
+     * @param name Name of device
+     */
+    public MediaDetails(SlotReference slotReference, CdjStatus.TrackType mediaType, String name) {
```

Comment:
This was one of the things that was missing to get the application to send requests to OpusProvider. When I filled this in MetadataFinder assumed the data was available and asked for Metadata from OpusProvider.

Reply:
That makes sense, we should probably make it package-private though?